### DISCUSSION TITLE
How-to track if specific content was viewed by visitor

### DISCUSSION THREAD

**Discussion by mheland** [plausible/analytics#2636] [Anything Else]
I was looking for a way to measure CTR Click Through Rate on a Call-To-Action image or text, I needed to know if the image/text was visible to the viewer, scrolled into view.

Found the Intersection Observer API on https://developer.mozilla.org/en-US/docs/Web/API/Intersection_Observer_API

Basically set a ID in a div somewhere on the page and trigger a Plausible custom event when this the div comes into view.

```
<div id="myCTA">
   <img src="cta.jpg" >
</div>
```

and then...

```
<script>
    console.log('Tracking CTA'); 

    const options = {
     threshold: 1,
    };

    function handleIntersection(entries) {
        entries.map((entry) => {
            if (entry.isIntersecting) {
            console.log('CTA visible - log event and unobserve');

            plausible('ctaVisible');

            observer.unobserve(entry.target);
            }
        });
        }

    const observer = new IntersectionObserver(handleIntersection, options);

    observer.observe(document.getElementById("myCTA"));

</script>

```
The `threshold ` option is for how much of the element that must be visible for the event to trigger, 1=100%

This should be in the docs - happy to write a more detailed how-to in that case.

**Comment by metmarkosaric** [comment]
thank you for sharing this guide @mheland! very useful! tagging it here with the word impressions in case someone else searches for this

**Reply by mheland** [reply]
Thank you! (Two years late, alrite!) @metmarkosaric - tried Plausible with React, it works like a dream, we track events on a SPA, had Claude plug it in for us, effortless the whole thing.

**Comment by jrgong420** [comment]
that for sharing, I am trying to achieve the same!

Is `ctaVisible` the name of the custom event then? @mheland

**Reply by mheland** [reply]
Yes, that's the custom event name, ctaVisible, define in `plausible('ctaVisible');`
And add it to Plausible Admin, Goals
